### DISCUSSION TITLE
Handler for live client don't seem to be configured as intended for use

### DISCUSSION THREAD

**Discussion by Nikhilesh-B** [deepgram/community#509] [General help]
### Which Deepgram product are you using?

Deepgram SDKs

### Details

I'm using the Deepgram's live client in the following way: 

```
`async def connect(self):
        print("We've connected to the websocket")
        self.sentence = ''
        try:
            self.deepgramLive = self.deepgram_client.listen.live.v("1")
            options = LiveOptions(smart_format=True,
                                  model="phonecall",
                                  tier="nova",
                                  language="en-US",
                                  encoding="mulaw",
                                  sample_rate=8000,
                                  endpointing="2000",
                                  utterance_end_ms="1000",
                                  interim_results=True)

            self.deepgramLive.start(options=options)
        except Exception as e:
            print(f'Could not open socket: {e}')
            return

        self.deepgramLive.on(LiveTranscriptionEvents.Error, self.on_error)
        self.deepgramLive.on(
            LiveTranscriptionEvents.Metadata, self.on_metadata)
        self.deepgramLive.on(
            LiveTranscriptionEvents.UtteranceEnd, self.on_utterance_end)
        self.deepgramLive.on(LiveTranscriptionEvents.Close, self.on_close)
        self.deepgramLive.on(
            LiveTranscriptionEvents.Transcript, self.on_message_simple1)

        await self.accept()`
```
        
        Now I have fixed the handler to be correctly work under the following format: 

        `def on_message_simple1(self, *args, **kwargs):
        if 'result' in kwargs:
            result = kwargs['result']
            if result is None:
                return
            self.sentence += result.channel.alternatives[0].transcript`
            
            But this is not the general function signature that is specified in the docs of the python sdk: 
            
            `def on_message(result=None):
                    if result is None:
                        return
                    sentence = result.channel.alternatives[0].transcript
                    if len(sentence) == 0:
                        return
                    print(f"speaker: {sentence}")`
    
    Under the hood these function handlers are invoked as follows in the LiveClient: 
    
`def _emit(self, event, *args, **kwargs):
        for handler in self._event_handlers[event]:
            handler(self, *args, **kwargs)
            
  def _listening(self) -> None:
      self.logger.debug("LiveClient._listening ENTER")
    
      while True:
          try:
              self.lock_exit.acquire()
              myExit = self.exit
              self.lock_exit.release()
              if myExit:
                  self.logger.notice("_listening exiting gracefully")
                  self.logger.debug("LiveClient._listening LEAVE")
                  return
    
              message = self._socket.recv()
              if len(message) == 0:
                  self.logger.info("message is empty")
                  continue
    
              data = json.loads(message)
              response_type = data.get("type")
              self.logger.verbose("response_type: %s", response_type)
    
              match response_type:
                  case LiveTranscriptionEvents.Transcript.value:
                      self.logger.debug(
                          "response_type: %s, data: %s", response_type, data
                      )
                      result = LiveResultResponse.from_json(message)
                      if result is None:
                          self.logger.error("LiveResultResponse.from_json is None")
                          continue
                      self._emit(
                          LiveTranscriptionEvents.Transcript,
                          result=result,
                          **dict(self.kwargs),
                      )`
    
    
    If I were to replicate the above code I would get the following error: 
    
    1. TypeError: TranscriptionConsumer.on_message() got multiple values for argument 'result', which I think is resulting from both result in the original function signature and results in the kwargs being passed to the handler; these are two parameters with the same name that the function doesn't know what to do with. 
    
    Is there another cause of this issue. I wasn't able to run the test code provided: `deepgram = DeepgramClient()

# Create a websocket connection to Deepgram
options = LiveOptions(
    punctuate=True,
    language="en-US",
    encoding="linear16",
    channels=1,
    sample_rate=16000,
)

def on_message(result=None):
    if result is None:
        return
    sentence = result.channel.alternatives[0].transcript
    if len(sentence) == 0:
        return
    print(f"speaker: {sentence}")

def on_metadata(metadata=None):
    if metadata is None:
        return
    print(f"\n{metadata}\n")

def on_error(error=None):
    if error is None:
        return
    print(f"\n{error}\n")

dg_connection = deepgram.listen.live.v("1")
dg_connection.start(options)

dg_connection.on(LiveTranscriptionEvents.Transcript, on_message)
dg_connection.on(LiveTranscriptionEvents.Metadata, on_metadata)
dg_connection.on(LiveTranscriptionEvents.Error, on_error)

# create microphone
microphone = Microphone(dg_connection.send)

# start microphone
microphone.start()

# wait until finished
input("Press Enter to stop recording...\n\n")

# Wait for the microphone to close
microphone.finish()

# Indicate that we've finished
dg_connection.finish()

print("Finished")`

because my pyaudio doesn't work.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by davidvonthenen** [comment]
The documentation should reflect this usage:
https://github.com/deepgram/deepgram-python-sdk/blob/main/examples/streaming/microphone/main.py#L33

If it doesn't, please link me to the incorrect docs and I can update them.
